spec/serializers/provider_serializer_spec.rb

```diff
@@ -84,4 +84,18 @@
 
     it { should eq provider.ucas_preferences.send_application_alerts_before_type_cast }
   end
+
+  describe 'contacts' do
+    let(:provider) do
+      create :provider,
+             contacts: [utt]
```

Comment:
do we care its an `utt` if not just `contact`

same with creating it `create(:contact)` as that is just `admin` which is now the default

then this bit can just be refactored as 
`|c| c[:type] == contact.type`

which altogether goes back to 
`describe 'contacts'` 

the subject in question is being able to find a thing of `type` 

where as 
```
  attribute :contacts do
    generate_provider_contacts + application_alert_recipient
  end```

so what is being tested, finding a single item from an array? as there is no other items in array or we dont care about `application_alert_recipient`

Reply:
How's this @defong? Better or worse?
```
describe 'contacts' do
    describe 'generate_provider_contacts returns the correct values' do
      let(:provider) do
        create :provider,
               contacts: [contact]
      end
      let(:contact) { create(:contact) }

      subject { serialize(provider)['contacts'].find { |c| c[:type] == contact.type } }

      its([:name]) { should eq contact.name }
      its([:email]) { should eq contact.email }
      its([:telephone]) { should eq contact.telephone }
    end

    describe 'the application alert recipient has been serialized upon instantiation of the contact object' do

      subject { serialize(provider)['contacts'].find { |c| c[:type] == 'application_alert_recipient' } }

      its([:name]) { should eq '' }
      its([:email]) { should eq provider.ucas_preferences.application_alert_email }
      its([:telephone]) { should eq ''}
    end
  end

```

Reply:
so it was 2 things 👍

Reply:
Yea it is added when the contacts serializer instantiates the contact object. generate_provider_contacts grabs each contacts name, type email etc. from provider.contacts. Then the application_alert_recipient is added separately.